tests/PHPStan/Type/Constant/ConstantStringTypeTest.php

```diff
@@ -148,4 +148,9 @@ public function testGeneralize(): void
 		$this->assertSame('class-string', (new ConstantStringType('NonexistentClass', true))->generalize()->describe(VerbosityLevel::precise()));
 	}
 
+	public function testInvalidEncoding(): void
+	{
+		$this->assertIsString((new ConstantStringType(file_get_contents('invalidUtf8Characters.txt', true)))->describe(VerbosityLevel::value()));
```

Comment:
I cannot use assertEquals or assertSame due I don't have the same string. Which assert would you use in this situation?

Reply:
assertSame with the exact string produced in the test